airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -214,27 +211,40 @@ private static Container getMain(final String image,
     return containerBuilder.build();
   }
 
-  private static void copyFilesToKubeConfigVolume(final ApiClient officialClient,
-                                                  final String podName,
+  private static void copyFilesToKubeConfigVolume(final String podName,
                                                   final String namespace,
                                                   final Map<String, String> files) {
     final List<Map.Entry<String, String>> fileEntries = new ArrayList<>(files.entrySet());
 
     // copy this file last to indicate that the copy has completed
     fileEntries.add(new AbstractMap.SimpleEntry<>(SUCCESS_FILE_NAME, ""));
 
+    Path tmpFile = null;
     for (final Map.Entry<String, String> file : fileEntries) {
       try {
+        tmpFile = Path.of(IOs.writeFileToRandomTmpDir(file.getKey(), file.getValue()));
+
         LOGGER.info("Uploading file: " + file.getKey());
-        final var contents = file.getValue().getBytes(StandardCharsets.UTF_8);
         final var containerPath = Path.of(CONFIG_DIR + "/" + file.getKey());
 
-        // fabric8 kube client upload doesn't work on gke:
-        // https://github.com/fabric8io/kubernetes-client/issues/2217
-        final Copy copy = new Copy(officialClient);
-        copy.copyFileToPod(namespace, podName, INIT_CONTAINER_NAME, contents, containerPath);
-      } catch (final IOException | ApiException e) {
+        // using kubectl cp directly here, because both fabric and the official kube client APIs both have
+        // several issues with copying files. See https://github.com/airbytehq/airbyte/issues/8643 for
+        // detauls.
+        final String command = String.format("kubectl cp %s %s/%s:%s -c %s", tmpFile, namespace, podName, containerPath, INIT_CONTAINER_NAME);
+        LOGGER.info(command);
+
+        final Process proc = Runtime.getRuntime().exec(command);
+        LOGGER.info("Waiting for kubectl cp to complete");
+        proc.waitFor();
+        LOGGER.info("kubectl cp complete, closing process");
+        proc.destroy();
+
+      } catch (final IOException | InterruptedException e) {
         throw new RuntimeException(e);
+      } finally {
+        if (tmpFile != null) {
+          tmpFile.toFile().delete();
```

Comment:
We do we need something to periodically sweep these if the files are being deleted in this `finally` block? Are you saying we need this in case this finally block is not executed for some reason?

Reply:
> In the near future this won't be a concern for syncs on Kubernetes when they use the container orchestrator, but it still will for OSS.

This logic is only performed for kubernetes, not docker. Are you saying this won't be a concern for cloud?

Reply:
🤦 sorry, yes. This isn't a concern at all.